tools/autograd/derivatives.yaml

```diff
@@ -815,11 +815,11 @@
   result: auto_element_wise
 
 - name: igamma(Tensor self, Tensor other) -> Tensor
```

Comment:
for the jvp errors and all that, you can add `result: auto_element_wise` in `igamma` and `igammac` to autogen the jvp for these.

Reply:
Alas: `RuntimeError: Derivative definition of igamma in derivatives.yaml defines the forward definition of gradient as element_wise but this only works for functions with a single differentiable input and a single differentiable output.`

Reply:
Righ, Then do `restult: igamma(grad, other) + igamma(self, other)`. This should be enough as we don't implement complex numbers yet. If we did we'd need to add a conj as well.

Reply:
Sorry... I'm not following. Did you mean `igamma(grad, other) + igamma(self, grad)` or, in fact, what you wrote? I'm not following the formula, and know it's simple calculus. I'm also a *little* foggy on what `grad` even is, which doesn't help.

Reply:
I'm sorry, the formula above should be:
```python
result: self_t * igamma_self_backward(self_p, other_p) -other_t * exp((self_p - 1) * log(other_p) - other_p - lgamma(self_p))
```
where `*_t` stands for tangent, and `*_p` stands for primal.

There is a short explanation of the unidimensional case at
https://github.com/pytorch/pytorch/blob/1439bd3c9c0a48d526bc20021486266591c0f480/tools/autograd/load_derivatives.py#L332-L342
the multidimensional case follows from a similar observation + the chain rule, or simply thinking about the multidimensional jacobian as a larger matrix a performing the computations above by blocks.

You can find similar applications of this formula at
https://github.com/pytorch/pytorch/blob/1439bd3c9c0a48d526bc20021486266591c0f480/tools/autograd/derivatives.yaml#L790-L799

As for the obvious questions "why doesn't `auto_element_wise` support multiple inputs?"... well, no one implemented codegen for it yet :)